rust/src/x509/mod.rs

```diff
@@ -79,6 +94,34 @@ pub unsafe extern "C" fn rs_x509_get_subject(ptr: *const X509) -> *mut c_char {
     rust_string_to_c(subject)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rs_x509_get_subjectaltname_len(ptr: *const X509) -> u16 {
+    if ptr.is_null() {
+        return 0;
+    }
+    let x509 = cast_pointer! {ptr, X509};
+    let san_list = x509.0.tbs_certificate.subject_alternative_name();
+    if let Ok(Some(sans)) = san_list {
+        return sans.value.general_names.len() as u16;
+    }
+    return 0;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_x509_get_subjectaltname_at(ptr: *const X509, idx: u16) -> *mut c_char {
+    if ptr.is_null() {
+        return std::ptr::null_mut();
+    }
+    let x509 = cast_pointer! {ptr, X509};
+    let san_list = x509.0.tbs_certificate.subject_alternative_name().unwrap();
+    if let Some(sans) = san_list {
+        let general_name = &sans.value.general_names[idx as usize];
+        let dns_name = SCGeneralName(general_name);
```

Comment:
Why do we need this conversion to `SCGeneralName` ?

Reply:
Because, the outputs are of the format `DNSName(www.suricata.io)` i.e. wrapped in the type that the entity is.
But, we only need to parse the actual entity and save it for use in sticky buffer. In this case, for example, `www.suricata.io`

Reply:
Does not DNSName have a to_string function ?

Reply:
The type is `GeneralName` and yes, it does not have a `to_string` fn. `to_string` on a `GeneralName` type yields `"DNSName(www.suricata.io)"` instead of its usual `DNSName(www.suricata.io)`